Fabric.Authorization.Persistence.SqlServer/Stores/SqlServerRoleStore.cs

```diff
@@ -118,15 +142,101 @@ public Task<IEnumerable<Role>> GetRoles(string grain, string securableItem = nul
 
             if (!string.IsNullOrWhiteSpace(securableItem))
             {
-                roles = roles.Where(p => string.Equals(p.SecurableItem.Name, securableItem));
+                roles = roles.Where(r => string.Equals(r.SecurableItem.Name, securableItem));
             }
 
             if (!string.IsNullOrWhiteSpace(roleName))
             {
-                roles = roles.Where(p => string.Equals(p.Name, roleName));
+                roles = roles.Where(r => string.Equals(r.Name, roleName));
+            }
+
+            return roles;
+        }
+
+        public async Task<Role> AddPermissionsToRole(Role role, ICollection<Permission> permissions)
+        {
+            var roleEntity = await _authorizationDbContext.Roles
+                .SingleOrDefaultAsync(r =>
+                    r.RoleId == role.Id
+                    && !r.IsDeleted);
+
+            if (role == null)
+            {
+                throw new NotFoundException<Role>($"Could not find {typeof(Role).Name} entity with ID {role.Id}");
+            }
+
+            var permissionIds = permissions.Select(p => p.Id);
+            var permissionDictionary = _authorizationDbContext.Permissions
```

Comment:
Do we need to run this query here, we already have the PermissionId's from the service?

Reply:
We have the PermissionIds, which is a set of Guids. I need to get the Sql Server `int` ID to create the relationship.

Reply:
OK - If having to requery to make an association is a side effect, I think we need to consider the idea of getting rid of the Sql Server identity column as an int. We don't have to consider it for this PR, but perhaps in another PR.

Reply:
Yeah - lets add a work item to the PBI to investigate and potentially make that change across the entities.